src/stores/SearchStore.ts

```diff
@@ -120,20 +120,18 @@ export class SearchStore {
 	@observable messagesFilterInfo: SSEFilterInfo[] = [];
 
 	@computed get searchProgress() {
-		const lastItem = this.currentSearch?.results[this.currentSearch.results.length - 1];
-		const lastItemTimestamp = lastItem
-			? getTimestampAsNumber(isEventNode(lastItem) ? lastItem.startTimestamp : lastItem.timestamp)
-			: null;
 		return {
 			startTimestamp: this.searchForm.startTimestamp,
 			endTimestamp: this.searchForm.endTimestamp,
-			currentPoint: lastItemTimestamp
-				? lastItemTimestamp - Number(this.searchForm.startTimestamp)
+			currentPoint: this.currentSearchTimestamp
```

Comment:
we need to store this data in currentSearch, otherwise loading state stays the same event if you are switching history